flow360/component/flow360_params/solvers.py

```diff
@@ -561,11 +550,13 @@ class NavierStokesSolverLegacy(NavierStokesSolver, LegacyModel):
     pressure_correction_solver: Optional[PressureCorrectionSolver] = pd.Field(
         alias="pressureCorrectionSolver"
     )
-    linear_solver_config: Optional[LinearSolverLegacy] = pd.Field(
-        alias="linearSolverConfig", default=LinearSolverLegacy()
-    )
+    linear_solver_config: Optional[LinearSolverLegacy] = pd.Field(alias="linearSolverConfig")
     linear_iterations: Optional[PositiveInt] = pd.Field(alias="linearIterations")
 
+    _processed_linear_solver_config = pd.validator(
+        "linear_solver_config", allow_reuse=True, pre=True
+    )(set_linear_solver_config_if_none)
+
```

Comment:
if we just want to set defaults, why we don't do:
```python
linear_solver_config: Optional[LinearSolverLegacy] = pd.Field(default=LinearSolverLegacy(), alias="linearSolverConfig")
```
?

Reply:
Because defaults are applied before Pre=True validator.

Once the default LinearSolverLegacy() gets applied then the new validator will never take any action as the `linearSolverConfig` is no longer None.